hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/resolver/MountTableResolver.java

```diff
@@ -381,18 +417,37 @@ public void clear() {
   public PathLocation getDestinationForPath(final String path)
       throws IOException {
     verifyMountTable();
+    PathLocation res;
+    String tmpPath = path;
+    if (isTrashPath(tmpPath)) {
+      tmpPath = subtractBaseTrashPath(tmpPath);
+    }
+    String finalPath = tmpPath;
     readLock.lock();
     try {
       if (this.locationCache == null) {
-        return lookupLocation(path);
+        res = lookupLocation(finalPath);
+      } else {
+        Callable<? extends PathLocation> meh = new Callable<PathLocation>() {
```

Comment:
We should have something better than meh, and this is an ideal candidate for using a lambda.

Reply:
It was there already but let's fix it :)